### DISCUSSION TITLE
wait for stream

### DISCUSSION THREAD

**Discussion by brixxnatt** [ant-media/Ant-Media-Server#6617] [General]
Sometime  we have a down time for 5 seconds on the internet , we need the  AMS to wait  , not cut the recording , is there a way to do that , please can someone explain  ,  AMS latest Enterprise . thanks

**Comment by SelimEmre** [comment]
Hi @brixxnatt,

How can you send your stream(WebRTC or RTMP)? Could you please give me more detail about your use case scenario?

**Reply by brixxnatt** [reply]
RTMP

**Reply by SelimEmre** [reply]
Thanks for the details @brixxnatt. I'm using macOS, so I needed to check the OBS settings on Windows. I have some advice for you:

1. You can add an RTMP ingest buffer to the Ant Media Server application settings directly from the Ant Media Server Dashboard. This option allows Ant Media Server to add a buffer to your RTMP streams, ensuring that stream recording continues even if there is a disconnection in your stream.

```
settings.rtmpIngestBufferTimeMs=6000
```

This setting will add a 6-second buffer to your stream. You can adjust them according to your requirements.

2. You can also add a buffer or increase the keyframe value in your OBS or RTMP settings.

For macOS OBS settings:
```
    Go to Settings > Output.
    Ensure you are in Advanced Output Mode.
    Under Streaming, make sure the Keyframe Interval is set to 2 seconds. This helps the stream recover more quickly after interruptions.
    Set the Bitrate to a level your network can handle comfortably, even during brief disruptions. For example, if your upload speed is 10 Mbps, consider setting the bitrate to around 5-6 Mbps to provide a buffer.
```

For Windows OBS settings:
Increasing the buffering time can help smooth out brief interruptions by allowing OBS to cache more video data.
```
    Go to Settings > Output.
    Switch to Advanced Output Mode.
    Under Streaming, find Network settings.
    Increase the Buffering Time. Setting this to around 500ms or higher can help smooth over network hiccups.
```

I hope this helps with your issue.

**Reply by yashtandon113** [reply]
Thanks, @SelimEmre for answering the question 👍 

@brixxnatt Please check the @SelimEmre's answer here.

**Reply by brixxnatt** [reply]
thanks for the answer , but it seams not working , when the stream is interrupsfor 5 seconds the recorder stop from the 1st second
do i need to restart AMS when I add these settings , I even put it to 60000 , still the recorder stop instantly

**Reply by SelimEmre** [reply]
Which version of Ant Media Server are you using? In the latest versions, you can directly add or change settings from the Management Panel under Advanced Settings. There's no need to restart AMS anymore.

![image](https://github.com/user-attachments/assets/1537cdae-17e4-4ebc-9e64-0fe68fe2ba52)

<img width="1779" alt="Screenshot 2024-08-29 at 18 00 50" src="https://github.com/user-attachments/assets/c056e9f5-2ebd-4d39-aac1-887eaa8fddfc">

**PS**: You can check your app settings directly using the REST API, as shown in the attached image. Please note that when you change the app settings, you need to restart the broadcasts for the changes to take effect. Also, make sure that you're using the correct application :)

<img width="1786" alt="Screenshot 2024-08-29 at 18 03 16" src="https://github.com/user-attachments/assets/22576587-0579-4a18-9fcb-68c713cead92">

**Reply by SelimEmre** [reply]
> thanks for the answer , but it seams not working , when the stream is interrupsfor 5 seconds the recorder stop from the 1st second do i need to restart AMS when I add these settings , I even put it to 60000 , still the recorder stop instantly

Are you still having problems? Please let me know if you're still getting an error @brixxnatt

**Comment by brixxnatt** [comment]
no one??? should I contact support ???

**Reply by muratugureminoglu** [reply]
Please check @SelimEmre message the above one.

**Comment by brixxnatt** [comment]
this is still need some support plz

**Reply by filip-ams** [reply]
Hi @brixxnatt, have you had a chance to try @lastpeony's suggestion? If you're still facing issues after switching to HLS recording, could you please share a screenshot of your application setup so we can better assist you?

**Comment by lastpeony** [comment]
I would disable mp4 recording and rely on HLS recording.

You can generate MP4 files through hls files(m3u8-ts) using ffmpeg on your side easily.

This way in such failure scenarios(including crash) you dont lose anything. If you write hls files directly to S3 bucket its even better.
